CONTRIBUTING.md

```diff
@@ -1,13 +1,115 @@
 This guide explains how to contribute to cocotb, and documents the processes we agreed on to manage the project.
 All processes in this document are designed to streamline the development effort, to avoid bottlenecks, and to ultimately give a pleasant experience to all involved.
 
-Architecture and Scope of Cocotb
---------------------------------
+
+## Setting Up a Development Environment
+
+Assuming you have used cocotb prior to reading this guide, you will already have the cocotb [installation prerequisites](https://docs.cocotb.org/en/latest/install.html) and standard development tools (editor, shell, git, etc.) installed.
+
+Additionally, you will need [doxygen](https://www.doxygen.nl/index.html), for building documentation, and [tox](https://pypi.org/project/tox/), for building documentation and running regression tests.
+
+We recommend if you are using a Linux distribution to use your system package manager to install doxygen.
+Likewise, doxygen can be installed using the homebrew package manager on Mac OS.
+Windows contributors should download a binary distribution installer from the main website.
+
+`tox` is a Python project and can be installed with `pip`.
+
+```command
+pip install tox
+```
+
+Finally, you should [fork and clone](https://guides.github.com/activities/forking/) the cocotb repo.
+This will allows you to make changes to cocotb source code, and run regressions and build documentation locally.
+
+Now you are ready to contribute!
+
+
+## Running Tests Locally
+
+First, [set up your development environment](#setting-up-a-development-environment).
+
+Our tests are managed by `tox`, which runs both `pytest` tests and our system of makefiles.
+The regression does not end on the first failure, but continues until all tests in the `/tests` and `/examples` directories have been run.
+
+To run the tests locally with `tox`, you will need to select an appropriate test environment.
+Valid test environments are formatted as `{your python version}-{your OS}`.
+Valid python version values are `py35`, `py36`, `py37`, `py38`, or `py39`;
+and valid OS values are `linux`, `macos`, or `windows`.
+For example, a valid test environment is `py38-linux`.
+You can see the list of valid test environments by running the below command.
+
+```command
+tox -l
+```
+
+Once you know the test environment you wish to use, call `tox` .
+
+```command
+tox -e py38-linux
+```
+
+At the end of the regression, if there were any test failures, the tests that failed will be printed.
+Otherwise, tox will print a green `:)`.
+
+### Selecting a Language and Simulator for Regression
+
+`tox` supports the usage of the environment variables [`SIM`](https://docs.cocotb.org/en/stable/building.html#var-SIM) and [`TOPLEVEL_LANG`](https://docs.cocotb.org/en/stable/building.html#var-TOPLEVEL_LANG) to select a simulator and language to run the regression.
```

Comment:
```suggestion
`tox` supports the usage of the environment variables [`SIM`](https://docs.cocotb.org/en/latest/building.html#var-SIM) and [`TOPLEVEL_LANG`](https://docs.cocotb.org/en/latest/building.html#var-TOPLEVEL_LANG) to select a simulator and language to run the regression.
```
I'd use ``latest`` docs since this is about developing.